src/main/java/seedu/address/model/person/PersonDetailContainsKeywordAndExamPredicate.java

```diff
@@ -0,0 +1,97 @@
+package seedu.address.model.person;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LESSTHAN;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MORETHAN;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.model.exam.Exam;
+
+/**
+ * Tests that a {@code Person}'s details contains the keyword given.
+ */
+public class PersonDetailContainsKeywordAndExamPredicate implements Predicate<Person> {
+    private final Prefix prefix;
+    private final String keyword;
+    private final Exam exam;
+
+    /**
+     * Constructor for the PersonDetailContainsKeywordPredicate.
+     * @param prefix The prefix to indicate the detail of the person to be searched.
+     * @param keyword The keyword to be searched.
+     */
+    public PersonDetailContainsKeywordAndExamPredicate(Prefix prefix, String keyword, Exam exam) {
+        this.prefix = prefix;
+        this.keyword = keyword;
+        this.exam = exam;
+    }
+
+    /**
+     * Tests if the person's details contain the keyword.
+     */
+    @Override
+    public boolean test(Person person) {
+        if (PREFIX_LESSTHAN.equals(prefix)) {
+            if (person.getScores().containsKey(exam)) {
+                return Double.parseDouble(person.getScores().get(exam).toString())
+                        < Double.parseDouble(keyword);
+            } else {
+                // Handle case when the selected exam is not found in the person's scores
+                return false;
+            }
+        } else if (PREFIX_MORETHAN.equals(prefix)) {
+            if (person.getScores().containsKey(exam)) {
+                return Double.parseDouble(person.getScores().get(exam).toString())
+                        > Double.parseDouble(keyword);
+            } else {
+                // Handle case when the selected exam is not found in the person's scores
+                return false;
+            }
+        } else {
+            // Code should not reach here
+            return false;
```

Comment:
POTENTIAL FLAW HERE: What if the person's score is equal to the users input? The code can in fact reach here. This wont cause an issue because it will always filter less than and more than anyway, but it might be a good idea to explicitly handle it.

Reply:
The false csae will not hit because the if cases check for the prescence of the lt/ or mt/ prefixes. Furthermore, this predicate class will not be created if the prefixes are neither of these prefixes.